src/client/Microsoft.Identity.Client/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -48,6 +54,7 @@ internal MsalAccessTokenCacheItem(
             RawClientInfo = response.ClientInfo;
             HomeAccountId = homeAccountId;
             OboCacheKey = oboCacheKey;
+            AdditionalCacheKeyComponents = cacheKeyComponents?.OrderBy(kvp => kvp.Key).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);

```

Comment:
[Blocking] This returns a dictionary. Dictionaries are not sorted by keys.

Reply:
resolved